actors/builtin/miner/miner_actor.go

```diff
@@ -1109,8 +1130,14 @@ func handleProvingPeriod(rt Runtime) {
 		// Vest locked funds.
 		// This happens first so that any subsequent penalties are taken from locked pledge, rather than free funds.
 		newlyVestedAmount := rt.State().Transaction(&st, func() interface{} {
-			newlyVestedFund, err := st.UnlockVestedFunds(store, rt.CurrEpoch())
+			vestingFunds, err := adt.AsArray(adt.AsStore(rt), st.VestingFunds)
+			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load vesting funds")
+			newlyVestedFund, err := st.UnlockVestedFunds(vestingFunds, rt.CurrEpoch())
 			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to vest funds")
+
+			st.VestingFunds, err = vestingFunds.Root()
+			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to write vesting funds")
+
```

Comment:
This is going to conflict with the #648. Would you mind either reverting this one, or hold off landing until the other does?

Reply:
👍 It's a simple change I'll hold off until the big one lands